src/discussions/posts/post/PostFooter.test.jsx

```diff
@@ -104,4 +92,24 @@ describe('PostFooter', () => {
     renderComponent({ ...mockPost, following: false });
     expect(screen.queryByRole('button', { name: /follow/i })).not.toBeInTheDocument();
   });
+
+  it('test like button when voteCount is zero', async () => {
+    renderComponent({ ...mockPost, voteCount: 0 });
+    expect(screen.queryByRole('button', { name: /like/i })).not.toBeInTheDocument();
+  });
+
+  it('test like button when voteCount is not zero', async () => {
+    renderComponent({ ...mockPost, voted: true, voteCount: 4 });
+    const likeButton = screen.getByRole('button', { name: /like/i });
+    await act(async () => {
+      fireEvent.mouseEnter(likeButton);
+    });
+
+    expect(screen.getByRole('tooltip')).toHaveTextContent(/unlike/i);
+    await act(async () => {
+      fireEvent.click(likeButton);
+    });
+    // clicking on the button triggers thread update.
+    expect(store.getState().threads.status === RequestStatus.IN_PROGRESS).toBeTruthy();
```

Comment:
please compare state after request is successful. 
await waitFor(() => {
      expect(screen.queryByRole('button', { name: /like/i })).not.toBeInTheDocument();
    });

Reply:
To compare state  thread needs to be updated explicitly here, which is not required here